core/src/data/table.rs

```diff
@@ -30,6 +31,7 @@ impl<'a> Table<'a> {
 
                 Ok(Self { name, alias, index })
             }
+            TableFactor::Derived { .. } => Err(Error::Table(TableError::Unreachable)),
```

Comment:
Interesting...
`data::Table` is a kind of... temporal data struct which is only for handling `TableFactor::Table`.
Now `TableFactor` enum has more than a single types.

`data::Table` also was being used in `join` and `select.
Now, we may not need `data::Table` anymore, then the code can be even quite more simple I think..